lib/net.js

```diff
@@ -736,9 +736,11 @@ Socket.prototype._getpeername = function() {
   if (!this._handle || !this._handle.getpeername) {
     return this._peername || {};
   } else if (!this._peername) {
-    this._peername = {};
+    const out = {};
+    const err = this._handle.getpeername(out);
     // FIXME(bnoordhuis) Throw when the return value is not 0?
-    this._handle.getpeername(this._peername);
+    if (err) return {};
```

Comment:
```suggestion
    if (err) return out;
```

Reply:
I fixed it. It seems mutated only on no-error.
https://github.com/nodejs/node/blob/8b8da00c39f6589752874153e115f60539eafa88/src/node_internals.h#L77-L79